lib/constable/models/user.ex

```diff
@@ -4,14 +4,14 @@ defmodule Constable.User do
   alias Constable.UserInterest
   alias Constable.Subscription
 
-  @permitted_email_domain Application.fetch_env!(:constable, :permitted_email_domain)
-
   defimpl Bamboo.Formatter do
     def format_email_address(user, _opts) do
       {user.name, user.email}
     end
   end
 
+  def permitted_email_domain, do: Application.fetch_env!(:constable, :permitted_email_domain)
```

Comment:
yeah, the environment variable is available at compilation time, so I think the issue is that the value of the variable may or not be there during compilation time. In production, you don't want to use module attributes with ENV variables because those variables might not be available during compilation. But in my experience, this problem is wonky, like, it sometimes works anyway, but in different environments, it may not.